packages/react-components/src/components/tooltip-component.tsx

```diff
@@ -1,121 +1,194 @@
+                    const weAreHoveringOverTooltip = isMouseOverRef.current;
+
+                    if (weAreHoveringOverTooltip) {
+                        debouncedUpdate.cancel();
+                        return prevState;
+                    } else if (weAreHidingTooltip) {
+                        // Don't update the content
+                        updatedState.content = prevState.content;
+                        // Keep z-index high during fade out
+                        updatedState.zIndex = prevState.zIndex;
+                        return updatedState;
+                    } else {
+                        calculateAndSetPosition();
+                        // Update z-index immediately when showing
+                        if (newState.visible) {
+                            updatedState.zIndex = 99999;
                         }
-                    }, 500);
-                }
-            } else {
-                // content being fetched, hide the hourglass tooltip
-                ReactTooltip.hide();
+                        return updatedState;
+                    }
+                });
+            }, 500),
+        []
+    );
+
+    // Track mouse position and detect when mouse leaves viewport
+    useEffect(() => {
+        const handleMouseMove = (e: MouseEvent) => {
+            lastMousePosition.current = { x: e.clientX, y: e.clientY };
+        };
+
+        const handleMouseLeave = (e: MouseEvent) => {
+            // Check if the mouse has actually left the viewport
+            if (e.clientY <= 0 || e.clientY >= window.innerHeight || e.clientX <= 0 || e.clientX >= window.innerWidth) {
+                setState(prev => ({ ...prev, visible: false }));
             }
-        }
-        this.setState({ element, func });
-    }
+        };
+
+        document.addEventListener('mousemove', handleMouseMove);
+        document.addEventListener('mouseleave', handleMouseLeave);
+
+        return () => {
+            document.removeEventListener('mousemove', handleMouseMove);
+            document.removeEventListener('mouseleave', handleMouseLeave);
+        };
+    }, [debouncedUpdate]);
+
+    // Hide tooltip when scrolling
+    useEffect(() => {
+        const handleScroll = () => {
+            setState(prev => ({
+                ...prev,
+                visible: false
+            }));
+        };
+
+        // Use capture to catch all scroll events before they might be stopped
+        const options = { capture: true };
+
+        window.addEventListener('scroll', handleScroll, options);
+        document.addEventListener('scroll', handleScroll, options);
+
+        return () => {
+            window.removeEventListener('scroll', handleScroll, options);
+            document.removeEventListener('scroll', handleScroll, options);
+        };
+    }, []);
 
-    private getContent() {
-        if (this.state.content) {
-            return this.state.content;
+    // Handle content and visibility changes
```

Comment:
I am not sure where this happens or can be fixed, but when you hover over the tooltip and do a text range selection, some range selection remains when you close and reopen the tooltip, even when the text content changes.

Reply:
I don't know what you mean exactly...  Can you post a GIF?

Reply:
I hover on a state on the kworker row, then mouse over the tooltip and select the End time value:
![tooltip1](https://github.com/user-attachments/assets/d505a0c2-38e8-4cb2-8ec0-59f75402701a)

Then I move my mouse and hover over a state on the swapper row, and in the tooltip the End time value (a different string) is already selected:
![tooltip2](https://github.com/user-attachments/assets/d6a34b96-6308-4359-8a84-af89d127644c)

This example is neat, but when the tooltip content is very different, then the persisted selection can be anything, starting and ending in the middle of words from different properties.

Reply:
This has been addressed!